architecture/2025-06-compute-extensibility-feature-spec.md

```diff
@@ -0,0 +1,488 @@
+            ```
+    *   **Confidential Environment (`confi-env`):**
+        *   The platform engineer configures `confi-env` with specialized recipes for `Applications.Core/containers@2025-05-01-preview` that support deploying confidential containers (e.g., ACI Confidential Containers). These recipes will interpret a specific property on the container resource to provision confidential compute.
+        *   Example recipe registration (conceptual):
+            ```bash
+            rad recipe register confi-container-recipe --environment confi-env \
+              --resource-type Applications.Core/containers@2025-05-01-preview \
+              --template-path oci://ghcr.io/radius-project/recipes/core/aci-confidential-container:1.0.0
+            rad recipe register confi-redis-recipe --environment confi-env \
+              --resource-type Applications.Datastores/redisCaches \
+              --template-path oci://ghcr.io/radius-project/recipes/azure/redis:1.0.0 
+            ```
+            (Note: The Redis recipe might be the same if its deployment doesn't change based on the compute's confidentiality.)
+
+2.  **Developer: Define Application with Mixed Container Types and Connections**
+    *   The developer defines an application in a Bicep file (`app.bicep`). This definition includes:
+        *   A standard container (e.g., a frontend web server).
+        *   A confidential container (e.g., a backend service processing sensitive data), marked with a property like `confidential: true` (the exact property name and structure to be defined by the UDT schema).
+        *   A data store, like a Redis cache.
+        *   A connection from the confidential backend container to the Redis cache.
+    *   Example `app.bicep`:
+        ```diff
+        import radius as radius
+
+        @description('The Radius Application ID')
+        param application string
+
+        resource frontend 'Applications.Core/containers@2025-05-01-preview' = {
+          name: 'frontend'
+          properties: {
+            application: application
+            container: {
+              image: 'nginx:latest'
+              ports: {
+                web: {
+                  containerPort: 80
+                }
+              }
+            }
+            // This container is standard by default
+          }
+        }
+
+        resource backend 'Applications.Core/containers@2025-05-01-preview' = {
+          name: 'backend'
+          properties: {
+            application: application
+            container: {
+              image: 'mycorp/sensitive-processor:v1.2'
+              ports: {
+                api: {
+                  containerPort: 5000
+                }
+              }
+              env: {
+                REDIS_CONNECTION: cache.properties.connectionStrings.default
+              }
+            }
+            connections: {
+              cache: {
+                source: cache.id
+              }
+            }
+            extensions: [
```

Comment:
I think the need for extensions, aka "punch through", goes away with RRTs and recipes. If a platform engineering team needs platform-specific configuration they would implement it in the recipe and maybe provide the developers with configuration options in the RRT.